cdap-unit-test/src/test/java/co/cask/cdap/partitioned/PartitionConsumingTestRun.java

```diff
@@ -115,14 +118,22 @@ private void testWordCountOnFileSet(Function<ApplicationManager, ProgramManager>
     programManager = runProgram.apply(applicationManager);
     programManager.waitForFinish(5, TimeUnit.MINUTES);
 
+    List<RunRecord> runs = programManager.getHistory();
+    Assert.assertEquals(3, runs.size());
+    for (RunRecord run : runs) {
+      Assert.assertEquals(ProgramRunStatus.COMPLETED, run.getStatus());
+    }
+
     Assert.assertEquals(new Long(3), getCount(serviceURL, "a"));
     Assert.assertEquals(new Long(3), getCount(serviceURL, "b"));
     Assert.assertEquals(new Long(3), getCount(serviceURL, "c"));
 
     DataSetManager<PartitionedFileSet> outputLines = getDataset("outputLines");
     Set<PartitionDetail> partitions = outputLines.get().getPartitions(PartitionFilter.ALWAYS_MATCH);
 
-    Assert.assertEquals(2, partitions.size());
+    // each of the three MapReduce runs produces an output partition (even if there's no input data)
+    // however, Worker run doesn't produce a new output partition if there's no new input partition
+    Assert.assertEquals(produceOutputPartitionEachRun ? 3 : 2, partitions.size());
```

Comment:
This was the assertion that was failing. The number of output partitions is no longer always 2. It can be 3, since when the MR runs, it produces an (empty) output partition, even if there's no input partition data. Previously, the MR run failed (https://issues.cask.co/browse/CDAP-2945).